python/sglang/srt/constrained/base_tool_cache.py

```diff
@@ -16,50 +16,93 @@
 """Base tool cache for constrained decoding tools."""
 
 import time
+from dataclasses import dataclass
+from threading import Event, Lock
+from typing import Any, Dict, Tuple
+
+
+@dataclass
+class MapEntry:
+    event: Event
+    value: Any
+
+    def __iter__(self):
+        return iter((self.event, self.value))
 
 
 class BaseToolCache:
+    enable: bool
+    cache: Dict[str, MapEntry]
+    metrics: Dict[str, Any]
+    lock_cache: Lock
+    lock_metrics: Lock
```

Comment:
Because `BaseToolCache` is not a `dataclass`, so these fields should go to `__init__`